contracts/proxy/src/contract.rs

```diff
@@ -234,14 +282,30 @@ pub fn authorize_guardian_or_multisig(
     sender: &CanonicalAddr,
 ) -> Result<(), ContractError> {
     match MULTISIG_ADDRESS.may_load(deps.storage)? {
-        // if multisig is set, ensure it's address equal to the caller address
+        // if multisig adrdess is set, check whether sender is equal to it
         Some(multisig_address) if multisig_address.eq(sender) => Ok(()),
-        Some(_) => Err(ContractError::IsNotMultisig {}),
-        // if multisig is not set, ensure caller address is guardian
+        // otherwise do guardian auth
         _ => ensure_is_guardian(deps, sender),
     }
 }
 
+/// Is used to authorize user or multisig contract
+pub fn authorize_user_or_multisig(deps: Deps, sender: &Addr) -> Result<(), ContractError> {
```

Comment:
Related to comment above, perhaps we can have it as either `user` or `guardians` which could be single guardian or multisig.

Reply:
Done here https://github.com/nymlab/vectis/pull/8/commits/4c9d4185dde859779451e4b062dff54bc9831673